### PR TITLE
Issue/8

### LINKED ISSUE #8: Help command doesn't work - swalled by filter, formatting cleanup.
Need to change the syntax so that when running 'environment help' it isn't handled by 'environment <filter>'. Change the syntax to support:
- environment show - to show all
- environment filter <string> - to filter on the set
- environment help

And add tests to verify the help works.

### PR DESCRIPTION
**Pull request by clanzen**
Fixes #8

### CODE REVIEW

File: src/scripts/env.js [hunk 1]
```diff
@@ -45,16 +47,38 @@ const SENSITIVE_DATA_ENV = ['PASSWORD', 'SALT', 'TOKEN', 'SECRET', 'KEY', 'ACCOU
...
-	// Regex matching entry point.
+	// Show help
 	// ------------------------------------------------------------------------
-	robot.respond(SHOW_ENV, {id: 'hubot.show.environment'}, (res) => {
-		// See if a variable name filter was provided.
-		let matchString = '';
-		if (res.match.length > 1) {
-			matchString = res.match[1].trim();
-		}
+	robot.respond(ENV_HELP, {id: 'hubot.show.environment'}, (res) => {
```
Comment (chambrid) [inline src/scripts/env.js]: Should this be `hubot.help.environment`?

File: src/scripts/env.js [hunk 2]
```diff
@@ -45,16 +47,38 @@ const SENSITIVE_DATA_ENV = ['PASSWORD', 'SALT', 'TOKEN', 'SECRET', 'KEY', 'ACCOU
...
+	// ------------------------------------------------------------------------
+	robot.respond(ENV_SHOW, {id: 'hubot.show.environment'}, (res) => {
+		robot.logger.debug(`${TAG}: hubot-environment, show all`);
+		processEnvRequest(res, '');
+	});
+
+	// ------------------------------------------------------------------------
+	// Show a filtered list of env variables.
+	// ------------------------------------------------------------------------
+	robot.respond(ENV_FILTER, {id: 'hubot.show.environment'}, (res) => {
```
Comment (chambrid) [inline src/scripts/env.js]: Should this be `hubot.filter.environment`?

Comment (clanzen) [inline src/scripts/env.js]: Done.

File: src/scripts/env.js (hunk 1 above)
Comment (clanzen) [inline src/scripts/env.js]: Done.

**Comment by chambrid** [comment]
👍

**Pull request merged**